tests/lib/nested.sh

```diff
@@ -211,23 +215,14 @@ create_nested_core_vm(){
             ;;
         ubuntu-20.04-64)
             NESTED_MODEL="$TESTSLIB/assertions/nested-20-amd64.model"
-
-            # shellcheck source=tests/lib/prepare.sh
-            . "$TESTSLIB"/prepare.sh
-            snap download --basename=pc-kernel --channel="20/edge" pc-kernel
-            uc20_build_initramfs_kernel_snap "$PWD/pc-kernel.snap" "$WORK_DIR/image"
```

Comment:
Shouldn't this be in an `if [ "$BUILD_SNAPD_FROM_CURRENT" = "true" ]` as well?
Once the earlier TODO is fixed at some point and there is an incompatible change snap-bootstrap/pc-kernel, the nested test would still run with snap-bootstrap from the distro package (whatever it will be at that point).

Reply:
Yes, this is comming in the next PR because but first I need to implement the re-sign of the kernel snap because currently it is not booting if I don't sign it after it is modified when running with secboot enabled. 
Claudio already explained me how to do it and my idea is to introduce that in a following PR.

Reply:
@bboozzoo I'll leave the conditional with default "false", does it make sense?
Thanks for the review